### PR TITLE
Modify `make` to not push to registry by default after build

### PR DESCRIPTION
**Pull request by lsago**
Since make will set command line variables to environment variables in subcommands, I use a slightly oddly named 'PUSH_REG' in case any command modifies behavior on 'PUSH' env var set.

This addresses https://github.com/vantage6/vantage6/issues/1191

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -5,10 +5,26 @@
...
+# Use `make PUSH_REG=true` to push images to registry after building
+PUSH_REG ?= false
+
+
+# We use a conditional (true on any non-empty string) later. To avoid
+# accidents, we don't use user-controlled PUSH_REG directly.
+# See: https://www.gnu.org/software/make/manual/html_node/Conditional-Functions.html
+_condition_push :=
+ifeq ($(PUSH_REG), true)
+	_condition_push := not_emtpy_so_true
```
Comment (bartvanb) [inline Makefile:25]: empty

**Review by bartvanb — APPROVED** [review]
one small spelling thing which would be nice to fix but not really necessary :-)

File: Makefile (hunk 1 above)
Comment (lsago) [reply Makefile]: so true! :)

**Comment by bartvanb** [comment]
I'll merge this into 4.4 as it is not too urgent

**Pull request merged**